utilities/scripts/python/diff.py

```diff
@@ -0,0 +1,156 @@
+from dataclasses import dataclass
+from typing import Dict
+
+
+@dataclass
+class DiffEntry:
+    """
+    A data class representing an entry in a difference collection.
+    Attributes:
+    val: any
+        The value of the entry.
+    add_or_remove_flag: bool
+        A flag indicating whether the entry should be removed (False) or added (True).
+    """
+
+    val: any
+    add_or_remove_flag: bool
+
+
+@dataclass
+class Diff:
+    """
+    Represents the difference between two dictionaries.
+    """
+    diff: dict
+
+    def __init__(self, expected: dict, provided: dict, strict: bool):
+        if strict:
+            self.diff = _strict_diff_(expected, provided)
+        else:
+            self.diff = _inclusion_diff_(expected, provided)
+
+    def has_diff(self) -> bool:
+        return True if self.diff else False
+
+    def to_ascii_colored_string(
+        self,
+        obj_name_to_add: str,
+        obj_name_to_remove: str,
+    ) -> str:
+        """
+        Generates an ascii colored string representation of the diff.
+
+        Args:
+            obj_name_to_add (str): The name of the object to add.
+            obj_name_to_remove (str): The name of the object to remove.
+        """
+
+        def _impl_(
+            diff: dict,
+            obj_name_to_add: str,
+            obj_name_to_remove: str,
+            ident: str = "",
+            path: str = "",
+        ) -> str:
+            result = ""
```

Comment:
move `_add_color_` function here, and then remove the `_` from its name.
Its visibility will be controlled by its scope.